File: Data Structures & Algorithms/two-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = new Map();
        for(let i=0; i<nums.length; i++){
            let x = target - nums[i];
            if(map.has(x)){
                return [map.get(x), i];
            }
            map.set(nums[i], i);
        }

        return [-1,-1];
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::set<int> set = {};

        for (int i=0; i<nums.size(); i++)
        {
            if (!set.contains(nums[i]))
            {
                set.insert(nums[i]);
            }
            else 
            {
                return true;
            }
        }

        return false;
    }
};